File: Remove All Occurrences of a Substring/Remove_All_Occurrences_of_a_Substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int ind = s.find(part);
        int n = part.size();
        while(ind != string::npos) {
            s.erase(ind,n);
            ind = s.find(part);
        }
        return s;
    }
};

File: Remove All Occurrences of a Substring/Remove_All_Occurrences_of_a_Substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder sb = new StringBuilder(s);
        int n = part.length();
        int index = sb.indexOf(part);
        while(index != -1) {
            sb.delete(index,index+n);
            index = sb.indexOf(part);
        }
        return sb.toString();
    }
}